README.md

```diff
@@ -139,6 +139,24 @@ updateJob = batchSQLClient.update(job_id, NEW_QUERY)
 cancelJob = batchSQLClient.cancel(job_id)
 ```
 
+**COPY queries**
+
+COPY queries allow you to use the [PostgreSQL COPY command](https://www.postgresql.org/docs/10/static/sql-copy.html) for efficient streaming of data to and from CARTO.
+
+```python
+from carto.sql import CopySQLClient
+
+copyClient = CopySQLClient(auth_client)
+
+# COPY FROM example
+query = 'COPY copy_example (the_geom, name, age) FROM stdin WITH (FORMAT csv, HEADER true)'
```

Comment:
Maybe caused by my ignorance, I see some details here that leave this open to confusion and errors:
- The query uses `FROM stdin` but the function name is `copyfrom_file_path`.
- The file is a .csv and then you also specify the `FORMAT`.
- The query is a `COPY`, and the function is, as well.

Can't Python code be less coupled to the query? Example:

```python
result = copy_client.copyfrom_file_path('copy_example', ['the_geom', 'name', 'age'])
```
By specifying the target table and the columns you can then compose the query. There could also be named parameters with good defaults. In my previous example, `FORMAT` is inferred from the extension and `HEADER` defaults to true, but it could become explicit if needed: 

```python
result = copy_client.copyfrom_file_path('copy_example', ['the_geom', 'name', 'age'], format='text', header=False)
```

Reply:
My goal is to provide a robust low level interface, that covers the not-so-easy parts of chunking and compression in a mostly pythonic way.

> The query uses FROM stdin but the function name is copyfrom_file_path

https://www.postgresql.org/docs/10/static/sql-copy.html

That's what PG can accept as a query. We cannot accept a FROM file cause that refers to files in the DB cluster filesystem.


> The file is a .csv and then you also specify the FORMAT.

There are more formats: `text` and `binary`.

> The query is a COPY, and the function is, as well.

Yes. COPY FROM/TO queries are different from others in a fundamental way (batch or not): they can send/receive (big)data other than queries/resultsets.

> Can't Python code be less coupled to the query?

In general, it can't. As you can see in the docs, the COPY queries can be pretty generic, in particular for COPY TO. Doing so would couple the python low level interface with a particular use case.

What you propose can be added on top of the existing API.

There's more on the scoping of this in the issue: https://github.com/CartoDB/carto-python/issues/83

Hope this sheds some light